src/common/options.cc

```diff
@@ -3042,6 +3042,50 @@ std::vector<Option> get_global_options() {
     .set_description("mclock anticipation timeout in seconds")
     .set_long_description("the amount of time that mclock waits until the unused resource is forfeited"),
 
+    Option("osd_mclock_cost_per_io_msec", Option::TYPE_UINT, Option::LEVEL_BASIC)
+    .set_default(0)
+    .set_description("Cost per IO in milliseconds to consider per OSD (overrides _ssd and _hdd if non-zero)")
+    .set_long_description("This option specifies the cost factor to consider in msec per OSD. This is considered by the mClockScheduler to set an additional cost factor in QoS calculations. Only considered for osd_op_queue = mClockScheduler")
+    .set_flag(Option::FLAG_RUNTIME),
+
+    Option("osd_mclock_cost_per_io_msec_hdd", Option::TYPE_UINT, Option::LEVEL_BASIC)
```

Comment:
I'm confused about the units of cost here - translating these from ms to IOPS gives very low numbers. How is the user supposed to know how to set this? If the user shouldn't need to set this, it should be advanced or dev.

Is it possible to derive these from the max capacity iops options instead?

Reply:
I think I understand where the issue is. I was using the overall max_osd_capacity in my calculations instead of the allocated capacity for the client class type leading to lower than expected performance. Therefore, at that point to mitigate this I introduced the cost per io config option to enable one to tune the performance. In any case, I have fixed the main issue. I will still retain the config option and make it 'dev' so that users cannot simply set this. And by default, this option will be disabled. If this option is unnecessary, I can remove it during the bug fix stage. Hopefully this should be okay?

Reply:
I see, that sounds good to me

Reply:
@jdurgin After fixing the calculation, I was still observing lower than expected performance when item costs went beyond 4MiB. For e.g. during backfilling I see the item costs consistently flap between 8MiB and 20MiB leading to high tag increment values. The tag increments between each request due to the scaling model gets high beyond 4 MiB item cost. To resolve this for now I have capped the item cost calculations at 4 MiB. That resolves the performance issue to an extent, but we either need benchmark data or better scaling model going forward. I have made a note of this in the code. I will look further into this in the coming days.

Reply:
Thanks for the explanation @sseshasa, that seems fine for now. We can continue fine-tuning things in future PRs.